File: src/iwnil_backend/src/main/java/ua/lpnu/lemoncat/iwnil/model/Game.java
package ua.lpnu.lemoncat.iwnil.model;

import lombok.Builder;
import lombok.Data;
import ua.lpnu.lemoncat.iwnil.data.AIType;
import ua.lpnu.lemoncat.iwnil.data.GameStatus;


@Data
@Builder
public class Game {
    private String firstPlayer;
    private String secondPlayer;
    private Integer nextMove;
    private GameStatus gameStatus;
    private AIType aiType;
    private int[][] board;
}

File: src/iwnil_hardware/src/game/Game.cpp
#include "game/Game.h"
#include "enum/Game_Status.h"

Game *Game::newGame(String firstPlayer, String secondPlayer, String aiType)
{
    return new Game(IN_PROGRES, firstPlayer, secondPlayer, 1, aiType);
}

Game::Game(String gameStatus,
           String firstPlayer,
           String secondPlayer,
           byte nextMove,
           String aiType)
{
    this->gameStatus = gameStatus;
    this->firstPlayer = firstPlayer;
    this->secondPlayer = secondPlayer;
    this->nextMove = nextMove;
    this->aiType = aiType;
    this->board = new Board(10, 10);
}

Game::Game(String gameStatus,
           String firstPlayer,
           String secondPlayer,
           byte nextMove,
           String aiType,
           Board *board)
{
    this->gameStatus = gameStatus;
    this->firstPlayer = firstPlayer;
    this->secondPlayer = secondPlayer;
    this->nextMove = nextMove;
    this->aiType = aiType;
    this->board = board;
}

Game::~Game()
{
    delete this->board;
}